lib/Bio/EnsEMBL/DataCheck/Checks/ExonsBoundaries.pm

```diff
@@ -0,0 +1,163 @@
+=head1 LICENSE
+
+Copyright [2018] EMBL-European Bioinformatics Institute
+
+Licensed under the Apache License, Version 2.0 (the 'License');
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an 'AS IS' BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+
+=cut
+
+package Bio::EnsEMBL::DataCheck::Checks::ExonsBoundaries;
+
+use warnings;
+use strict;
+
+use Moose;
+use Test::More;
+use List::Util qw[min max];
+
+extends 'Bio::EnsEMBL::DataCheck::DbCheck';
+
+use constant {
+    NAME        => 'ExonsBoundaries',
+    DESCRIPTION => 'Checks exon boundaries in their transcript (same strand / correct order / min start / max end)',
+    DB_TYPES    => [ 'core' ],
+    GROUPS      => [ 'core_handover' ],
+};
+
+sub tests {
+  my ($self) = @_;
+
+  my $mapper = sub {
```

Comment:
I'm not a fan of this approach. It necessitates the creation of lots of variables (most of which are counter-intuitively initialised to negative values) which then need to be tracked and incremented. I appreciate that this is how it was done in the HC, and that it works; it's just difficult to understand and maintain. Another disadvantage is that the logic for several tests is all bundled up together; if we wanted to split the functionality up, it'd be tricky to disentangle.

Reply:
My suggestion would be to copy the functionality from the https://github.com/Ensembl/ensembl-datacheck/blob/master/lib/Bio/EnsEMBL/DataCheck/Checks/GeneTranscriptBounds.pm datacheck. This does for genes and transcripts most of what needs to be done for transcripts and exons: there's a test for checking seq_region and strand consistency, and that the extremities match up.

That just leaves the exon overlap functionality. I imagine that could be done with a pretty hairy SQL query; but the existing approach here is probably better. Without all the other functionality, the code to check for overlaps is greatly simplified.